src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -27,22 +27,18 @@
  */
 class EntityUserProvider implements UserProviderInterface
 {
+    private $em;
+    private $classOrAlias;
+    private $property;
+
     private $class;
     private $repository;
-    private $property;
     private $metadata;
 
-    public function __construct(ManagerRegistry $registry, $class, $property = null, $managerName = null)
+    public function __construct(ManagerRegistry $registry, $classOrAlias, $property = null, $managerName = null)
     {
-        $em = $registry->getManager($managerName);
-        $this->class = $class;
-        $this->metadata = $em->getClassMetadata($class);
-
-        if (false !== strpos($this->class, ':')) {
-            $this->class = $this->metadata->getName();
-        }
-
-        $this->repository = $em->getRepository($class);
+        $this->em = $registry->getManager($managerName);
```

Comment:
even better: you should store the registry and the manager name instead of instantiating the entity manager here. This would also make it compatible with resetting the EM.